malaria-ios/Managers/AchievementManager.swift

```diff
@@ -0,0 +1,174 @@
+import Foundation
+
+/// Handles general achievement logic throughout the app.
+
+class AchievementManager {
+  static let sharedInstance = AchievementManager()
+  
+  var context: NSManagedObjectContext! =
+    CoreDataHelper.sharedInstance.createBackgroundContext()!
+  
+  private var achievements: [Achievement] {
+    let fetchRequest = NSFetchRequest(entityName: "Achievement")
+    
+    do {
+      let result = try context.executeFetchRequest(fetchRequest) as! [Achievement]
+      return result as [Achievement]
+    } catch let error as NSError {
+      Logger.Error("Could not fetch \(error), \(error.userInfo)")
+    }
+    return []
+  }
+  
+  /**
+   Each achievement has it properties that needs to be fulfilled
+   in order to unlock.
+   */
+  
+  private var properties: [AchievementProperty] {
+    let fetchRequest = NSFetchRequest(entityName: "AchievementProperty")
+    
+    do {
+      let result = try context.executeFetchRequest(fetchRequest) as! [AchievementProperty]
+      return result as [AchievementProperty]
+    } catch let error as NSError {
+      Logger.Error("Could not fetch \(error), \(error.userInfo)")
+    }
+    return []
+  }
+  
+  func getAchievements(withTag tag: String) -> [Achievement] {
+    return achievements.filter { $0.tag == tag }
+  }
+  
+  func defineProperty(name: String,
```

Comment:
rename to createAchivementProperty (IMO it is better)

Btw, shouldn't this receive a achievement as argument? As far as I can see in the dataModel, the achievementProperty is attached to a Achievement, therefore the body of this method is not consistent with the data model you created.